src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_common.py

```diff
@@ -304,6 +305,19 @@ def flexible_server_version_upgrade(cmd, client, resource_group_name, server_nam
                        "We recommend selecting PostgreSQL 13 or a later version for "
                        "all future operations.")
 
+    list_server_capability_info = get_postgres_server_capability_info(cmd, resource_group_name, server_name)
+    eligible_versions = list_server_capability_info['supported_server_versions'][str(current_version)]
+    if version not in eligible_versions:
+        error_message = ""
+        if len(eligible_versions) > 0:
+            error_message = "Server is running version {}. It can only be upgraded to the following versions: {} ".format(str(current_version), eligible_versions)
+        else:
+            error_message = "Server is running version {}. It cannot be upgraded to any higher version. ".format(str(current_version))
+
+        if "17" in eligible_versions:
```

Comment:
Do we really want to block upgrading to version 17 from CLI until 17 is Generally Available and we release a new version of CLI, that would allow upgrading to that version?